connection.js

```diff
@@ -226,6 +228,30 @@ TChannelConnection.prototype.setupHandler = function setupHandler() {
     }
 };
 
+TChannelConnection.prototype.writeToSocket =
+function writeToSocket(buf) {
+    var self = this;
+
+    if (self.socket._writableState.buffer.length >
+        MAX_PENDING_SOCKET_WRITE_REQ
+    ) {
+        self.logger.warn('resetting connection due to write backup',
+            self.extendLogInfo({
+                bufferLength: self.socket._writableState.buffer.length,
+                maxlen: self.socket._writableState.length
+            })
+        );
+
+        // NUKE THE SOCKET
+        self.resetAll(Error.SocketWriteFullError({
+            pendingWrites: self.socket._writableState.buffer.length
```

Comment:
why don't we just create this error object above, and then use it in the warn log?